src/shared/theme/builder.ts

```diff
@@ -46,7 +55,58 @@ export class ThemeBuilder {
     return this;
   }
 
+  addReferenceTokens(referenceTokens: ReferenceTokens): ThemeBuilder {
+    this.theme.referenceTokens = referenceTokens;
+
+    // Process reference tokens and add generated tokens to theme
+    if (referenceTokens.color) {
+      const generatedTokens = this.processReferenceTokens(referenceTokens.color);
+      this.theme.tokens = { ...this.theme.tokens, ...generatedTokens };
+    }
+
+    return this;
+  }
+
   build(): Theme {
     return this.theme;
   }
+
+  private processReferenceTokens(colorTokens: ColorReferenceTokens): TokenCategory<string, string> {
+    const generatedTokens: TokenCategory<string, string> = {};
+
+    Object.entries(colorTokens).forEach(([colorName, paletteInput]) => {
+      const palette = this.processColorPaletteInput(paletteInput);
+
+      // Add generated palette tokens with naming convention: colorPrimary50, colorPrimary600, etc.
+      Object.entries(palette).forEach(([step, value]) => {
+        const tokenName = `color${this.capitalize(colorName)}${step}`;
+        generatedTokens[tokenName] = value;
+      });
+    });
+
+    return generatedTokens;
+  }
+
+  // Right now just validates steps, but will also handle seed token color generation in a future PR
+  private processColorPaletteInput(input: ColorPaletteInput): ColorPaletteDefinition {
+    const validSteps = [
+      50, 100, 150, 200, 250, 300, 350, 400, 450, 500, 550, 600, 650, 700, 750, 800, 850, 900, 950, 1000,
+    ];
```

Comment:
How about

```js
for(let i = 50; i <= 1000; i+=50) {
   validSteps.push(i);
}
```

this would prevent silly bugs of missing one number in this row when typing manually

(currently no numbers missing, but why doing error prone way)